src/resource.c

```diff
@@ -568,24 +568,30 @@ coap_add_resource(coap_context_t *context, coap_resource_t *resource) {
   resource->context = context;
 }
 
+/*
+ * Input context is ignored, but param left there to keep API consistent
+ */
 int
 coap_delete_resource(coap_context_t *context, coap_resource_t *resource) {
-  if (!context || !resource)
+  if (!resource)
     return 0;
```

Comment:
So finally it won't mimic the behavior of `free` functions? Are there plans to add `coap_free_resource` which never fails?

Reply:
It does mimic the behavior of `free()` functions in the sense that if resource is not defined (i.e. NULL) it returns without causing a core dump.  It does not mimic `free()` functions in that a value is returned.  `coap_delete_resource()` is not forcing you to check the return value.

I am for function naming consistency, but realize that there are some _free_ with more _delete_ functions publically available.  I do not see the need at present for `coap_free_resource()` as well.

Reply:
The consistent naming scheme that has evolved so far is that if a name contains "free" it releases allocated storage (i.e., it mimics the `free()` for a previous `malloc()`). A "delete" is more like the operator `delete` in C++ in that it does necessary cleanup as well (and may free allocated resources, usually by calling a corresponding ...`_free_`...